oauth2_lib/fastapi.py

```diff
@@ -346,7 +349,7 @@ async def authorize(self, request: HTTPConnection, user_info: OIDCUserModel) ->
         opa_input = {
             "input": {
                 **(self.opa_kwargs or {}),
-                **user_info,
+                **(user_info or {}),
```

Comment:
This suggests that `user_info` can also be `None`, does the annotation need to be updated?

Reply:
I updated the annotation, but since this is a rare case, I'm not very happy. However, from a code perspective, we should make it optional.

Reply:
I don't think the default `None` value is necessary though?

Reply:
Okay, I will make it required again. However, I have mixed feelings. On one hand, I think it makes sense that the authorize function needs a user, but on the other hand, for a callback step API called by LSO, we don't have a real user. That said, it seems to be a rare case. :-/

Reply:
Well, I've read in some OIDC related documentation about a scenario where anonymous (so un-authenticated) access is allowed to specific resources, with a subset of allowed actions (i.e. only readonly). So this would support that usecase. It is indeed kind of an edge case, we ourselves don't have a need for it.

Either way, the default behavior right now in the OIDCAuth is to require authentication, so that's good. Also if one has an OPA policy that requires authenticated users, then sending empty user_info into OPA should return a negative result. I don't see any risk here. :)